app/uk/gov/hmrc/partnershipidentificationfrontend/api/controllers/JourneyConfigJsonReader.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.partnershipidentificationfrontend.api.controllers
+
+import play.api.libs.json.{JsValue, Reads}
+import uk.gov.hmrc.partnershipidentificationfrontend.api.controllers.JourneyConfigJsonReader.jsonKeys
+import uk.gov.hmrc.partnershipidentificationfrontend.models.PartnershipType.PartnershipType
+import uk.gov.hmrc.partnershipidentificationfrontend.models.{JourneyConfig, PageConfig}
+
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
In my experience "Json" read and write methods are often defined in companion objects. Would it be better to put this method in an Object JourneyConfig defined along with the case class JourneyConfig?

Reply:
Richard I think Guice will complain because it knows about java not scala. We use Guice as dependency injection framework to inject things in the constructor of the controller .... so no problem at all if we define as @Singleton. I promise ;-)